### ISSUE TITLE
Function ID and Endianness conversion in ARM

### ISSUE THREAD

**Issue by pabx06** [NationalSecurityAgency/ghidra#1548]
Hello.
is ghidra able to mach little-endian lib's signatures in a big-endian or mixed endian project ?

**Comment by ghidra1** [comment]
The FunctionID (FID) signature database files are processor architecture and library specific.  Ghidra provides FID signature databases for various versions of Visual Studio libraries for x86 (32-bit and 64-bit, little-endian).  If this capability is needed for other cases you will need to generate your own FID signature database files using the FunctionID plugin to create such databases from imported and analyzed libraries (see FunctionID help content for more information).

**Comment by pabx06** [comment]
for arm it will be nice to reverse the byte before hashing this way you get both big endian and little endian sig from one lib ... assuming that compiler backend generator do reverse bytes for function body &data except may be strings data

**Comment by ghidra1** [comment]
Need to examine endianess concern more closely

**Comment by pabx06** [comment]
shall i also add: the ability to restrict the function ID over a specific memory range and  or specific function tag . finer than a whole mem block
ie
- startAddress= 0x.....
- endAddress= 0x.....
or 
Function ID  on function  where TAG="something"

**Comment by ghidra1** [comment]
I don't think it makes sense to incorporate function tags into FID since they are not used in mainstream analysis.  Adding function-by-function constraints when a FID database is generated in bulk (i.e., batch processing) seems unrealistic.

**Comment by pabx06** [comment]
> I don't think it makes sense to incorporate function tags into FID since they are not used in mainstream analysis. Adding function-by-function constraints when a FID database is generated in bulk (i.e., batch processing) seems unrealistic.

usecase: load **openssl.fid** ID database and run it on function **taged CRYPTO**
or **zlib.fid** on function **tagged COMPRESSION**
